bus-mapping/src/state_db.rs

```diff
@@ -75,6 +75,7 @@ pub struct StateDB {
     state: HashMap<Address, Account>,
     access_list_account: HashSet<Address>,
     access_list_account_storage: HashSet<(Address, U256)>,
+    refund: u64,
```

Comment:
I think we should define (and document via comment) the lifespan of the StateDB.  Initially I considered "block" lifespan.  Now it contains `access_list_account`, `access_list_account_storage` and `refund` which have "transaction" lifespan right?
So I think there are two options:
1. Build a new StateDB for each tx
2. Reuse the StateDB for the entire block, and after every tx reset the "transaction" lifespan fields.

In option 1 we query geth with state accounts + keys between each tx.  In option 2 we do all queries together for the entire block.

Reply:
You are right, in current approach I forget to clear refund and access list before handling next tx. I think option 2 + tx lifespan fields clearing would be okay to use, let me fix this in this PR.

But if we want split different lifespan fields into different structs, we might put them into `TxContext`, which is dropped automatically after tx is parsed. Could we decide if to do this or not after this PR?

Reply:
> But if we want split different lifespan fields into different structs, we might put them into `TxContext`, which is dropped automatically after tx is parsed. Could we decide if to do this or not after this PR?

Yes! Let's decide the approach after this PR :)